manager/controlapi/extension.go

```diff
@@ -0,0 +1,33 @@
+package controlapi
+
+import (
+	"context"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// CreateExtension creates an `Extension` based on the provided `CreateExtensionRequest.Extension`
+// and returns a `CreateExtensionResponse`.
+// - Returns `InvalidArgument` if the `CreateExtensionRequest.Extension` is malformed,
+//   or fails validation.
+// - Returns an error if the creation fails.
+func (s *Server) CreateExtension(ctx context.Context, request *api.CreateExtensionRequest) (*api.CreateExtensionResponse, error) {
+	return nil, nil
```

Comment:
Better to return some error here than `nil, nil`, in case something happens and it takes a while before this code actually gets merged.

Reply:
done.